js/src/leaflet.js

```diff
@@ -16,6 +16,7 @@ require('leaflet-fullscreen');
 require('leaflet-transform');
 require('leaflet.awesome-markers');
 require('leaflet-search');
+require('protomaps-leaflet');
```

Comment:
for general purpose vector tile display, maplibre is better than protomaps-leaflet, is it possible to use that instead?

Reply:
I used maplibre in the initial commit, but it failed. I don't how to fix it, so I changed to back to protomaps-leaflet. See
https://github.com/jupyter-widgets/ipyleaflet/pull/1138/commits/92586cd15866651da55e9918bc37c605fdb72a7b#diff-ab455b4e55f1402a4baec5dd7dd302a20e715e7b3897e82cc2e4b6ac557baba8

Reply:
For references on how styling vector tiles works with ipyleaflet. https://ipyleaflet.readthedocs.io/en/latest/layers/vector_tile.html

Reply:
i ended up using protomaps-leaflet's json_style function to convert mapbox style specs to protomaps style

https://github.com/jtmiclat/ipyleaflet/commit/c101c87c55f2c0baa94d3af0d3267f850a6b5aa9 

I found trying to use maplibre is super hacky with overlays and for the case of ipyleaflet adds a large js dependency for an optional feature.

Reply:
the json_style only parses a small subset of mapbox style specs. I don't have the time or resources to bring that to a complete state (the spec is very complex - look at https://github.com/openlayers/ol-mapbox-style for a comparison), which is why I would rather remove the feature instead of confusing developers as to why their JSON styles don't work. I may re-evaluate this, but in general the state of protomaps-leaflet is frozen as the majority of use cases have moved to MapLibre. This could change if a company or institution wants to sponsor the continued development of the protomaps-leaflet engine, a good use case may be these notebooks.